go/tools/builders/nogo_change.go

```diff
@@ -0,0 +1,88 @@
+package main
+
+
+import (
+	"fmt"
+	"go/token"
+	"os"
+	"path/filepath"
+
+	"golang.org/x/tools/go/analysis"
+)
+
+// DiagnosticEntry represents a diagnostic entry with the corresponding analyzer.
+type DiagnosticEntry struct {
+	analysis.Diagnostic
+	*analysis.Analyzer
+}
+
+// An Edit describes the replacement of a portion of a text file.
+type Edit struct {
```

Comment:
How would users apply these fixes? I wonder whether we shouldn't just export a regular patch file instead, possibly one per analyzer in a tree artifact in case different analyzers produce patches that conflict with each other.

This was my previous attempt at exporting fixes as patches: https://github.com/bazelbuild/rules_go/commit/998b738105e4fcefcf432f5d53723826073691d2

Reply:
thanks for sharing, I agree your proposal is a more end-to-end solution. Will include this. 

We have a separate CLI command for patching, which shares some common part with your version:
internally, it reads the edits, sorts them, applies them in memory to get the 'after' file, then produces the patch file by comparing before/after, then applies the patch file.

It holds the edits for each analyzer separately, in the patch file, edits of all analyzers are merged. If they overlap, it would fail.

Reply:
addressed in the new commit. 

one caveat is that, currently it copies/pastes https://github.com/pmezard/go-difflib/blob/master/LICENSE under its permission of copyright. 

after one day of attempt, I could not get test passing due to the dependency missing of https://github.com/pmezard/go-difflib. Can we address this at last or in a separate PR?